packages/astro/src/container/index.ts

```diff
@@ -0,0 +1,468 @@
+	/**
+	 * @default {}
+	 * @description
+	 *
+	 * A subset of the astro configuration object.
+	 *
+	 * ## Example
+	 *
+	 * ```js
+	 * const container = await AstroContainer.create({
+	 * 	astroConfig: {
+	 * 		trailingSlash: "never"
+	 * 	}
+	 * });
+	 * ```
+	 */
+	astroConfig?: AstroContainerUserConfig;
+};
+
+type AstroContainerManifest = Pick<
+	SSRManifest,
+	| 'middleware'
+	| 'clientDirectives'
+	| 'inlinedScripts'
+	| 'componentMetadata'
+	| 'renderers'
+	| 'assetsPrefix'
+	| 'base'
+	| 'routes'
+	| 'assets'
+	| 'entryModules'
+	| 'compressHTML'
+	| 'trailingSlash'
+	| 'buildFormat'
+	| 'i18n'
+>;
+
+type AstroContainerConstructor = {
+	streaming?: boolean;
+	renderers?: SSRLoadedRenderer[];
+	config: AstroConfig;
+	manifest?: AstroContainerManifest;
+	resolve?: SSRResult['resolve'];
+};
+
+export class experimental_AstroContainer {
+	#pipeline: TestPipeline;
+	#config: AstroConfig;
+
+	/**
+	 * Internally used to check if the container was created with a manifest.
+	 * @private
+	 */
+	#withManifest = false;
+
+	private constructor({
+		streaming = false,
+		renderers = [],
+		config,
+		manifest,
+		resolve,
+	}: AstroContainerConstructor) {
+		this.#config = config;
+		this.#pipeline = TestPipeline.create({
+			logger: new Logger({
+				level: 'info',
+				dest: nodeLogDestination,
+			}),
+			manifest: createContainerManifest(renderers, config, manifest),
+			streaming,
+			serverLike: true,
+			renderers,
+			resolve: async (specifier: string) => {
+				if (this.#withManifest) {
+					return this.#containerResolve(specifier);
+				} else if (resolve) {
+					return resolve(specifier);
+				}
+				return specifier;
+			},
+		});
+	}
+
+	async #containerResolve(specifier: string): Promise<string> {
+		const found = this.#pipeline.manifest.entryModules[specifier];
+		if (found) {
+			return new URL(found, this.#config.build.client).toString();
+		}
+		return found;
+	}
+
+	/**
+	 * Creates a new instance of a container.
+	 * 
+	 * @param {AstroContainerOptions=} containerOptions
+	 */
+	public static async create(
+		containerOptions: AstroContainerOptions = {}
+	): Promise<experimental_AstroContainer> {
+		const {
+			astroConfig = ASTRO_CONFIG_DEFAULTS,
+			streaming = false,
+			renderers = [],
+		} = containerOptions;
+		const config = await validateConfig(astroConfig, process.cwd(), 'container');
+		const loadedRenderers =  await Promise.all(
+			renderers.map(async (renderer) => {
+				const mod = await import(renderer.serverEntrypoint);
+				if (typeof mod.default !== 'undefined') {
+					return {
+						...renderer,
+						ssr: mod.default,
+					} as SSRLoadedRenderer;
+				}
+				return undefined;
+			})
+		);
+		const finalRenderers = loadedRenderers.filter((r): r is SSRLoadedRenderer => Boolean(r));
+		
+		return new experimental_AstroContainer({ streaming, renderers: finalRenderers, config });
+	}
+
+	// NOTE: we keep this private via TS instead via `#` so it's still available on the surface, so we can play with it.
```

Comment:
What does "play with it" mean?

Reply:
I plan to use it to create some integration that leverages it. However, I don't know yet how an integration would look like, so I wanted to keep it "available" to us maintainers until I have a clear idea on how we can use it.

What should I change in the comment?

Reply:
I reworded the comment, let me know what you think